src/pages/journal/journal.ts

```diff
@@ -171,6 +178,64 @@ export class JournalPage extends BasePageComponent implements OnInit {
     }
   }
 
+  /**
+   * If state contains no cached data then get users search results
+   * otherwise get create slot data directly
+   * @param emission
+   */
+  generateSlotAndSearchResults = (emission: SlotItem[]) => {
+    const noCacheData: boolean = true;
+    if (noCacheData && !this.searchResultsAppRefs) {
+      return this.getSearchResults(emission);
+    }
+    this.createSlots(emission, this.searchResultsAppRefs);
+
+  }
+
+  /**
+   * Call endpoint to obtain users previously completed tests and pass to createSlots function
+   * for evaluation
+   * @param emission
+   */
+  getSearchResults = (emission: SlotItem[]) => {
+    const advancedSearchParams: AdvancedSearchParams = {
+      startDate: '',
+      endDate: '',
+      staffNumber: removeLeadingZeros('1234567'),
+      costCode: '',
+    };
+    forkJoin([
+      this.searchProvider.advancedSearch(advancedSearchParams),
+    ]).subscribe(
+      ([
+         searchResultsTemp,
+       ]) => {
+        this.searchResultsAppRefs = searchResultsTemp.map((res) => {
+          return res.applicationReference;
+        });
+        this.createSlots(emission, this.searchResultsAppRefs);
+      },
+      err => this.showError(err),
+    );
+  }
+
+  /**
+   * Search for slots appReference in returned search results
+   * @param slot
+   * @param references
+   */
+  hasSlotBeenTested(slot, references): boolean {
+    const applicationReference: ApplicationReference = {
+      applicationId: slot.slotData.booking.application.applicationId,
+      bookingSequence: slot.slotData.booking.application.bookingSequence,
+      checkDigit: slot.slotData.booking.application.checkDigit,
+    };
+
+    return references.some((reference) => {
+      return reference === parseInt(formatApplicationReference(applicationReference), 10);
+    });
+  }
```

Comment:
Another idea for this part of the solution would be to simply update the test statuses themselves. You can use the existing `SetTestStatusSubmitted` action to set the status of the submitted tests to `Submitted`.
Then if you call the `PersistTests` action this will save the new statuses to the cache for you.
This way the journal should naturally try update its view to display the activity code of the test rather than the "Start test" button. There might be some typing issues with this approach though.

Reply:
Haven't approached it this way, but happy to discuss this further if the solution that is now implemented seems significantly weaker than what you described.